vault/resource_kv_secret_v2.go

```diff
@@ -217,12 +217,33 @@ func kvSecretV2Read(_ context.Context, d *schema.ResourceData, meta interface{})
 	shouldRead := !d.Get("disable_read").(bool)
 
 	path := d.Id()
+	if path == "" {
+		return nil
+	}
 
 	if err := d.Set(consts.FieldPath, path); err != nil {
 		return diag.FromErr(err)
 	}
 
 	if shouldRead {
+		parsedPath := strings.Split(path, "/")
```

Comment:
Seems good, but shouldn't we always be setting some values e.g. `mount`, `metadata`, etc... ?

You may want to limit the number of substrings to 3, using `strings.SplitN()` instead of `strings.Split()`,  I am not sure if the `name` component is allowed to contain `/`'s.

Reply:
Good call! I noticed another test was failing because that test dealt with nested secrets (`name` dod contain `/`'s in that case), and using `SplitN` instead to limit the substrings fixes that. Thanks for the callout!

I also updated the code to always set all the fields that can be set without reading the secret itself (in case of `disable_read`, we won't detect the drift on any fields that are dependent on the secret being read)